client/src/components/advancedSearch/TaskFilter.tsx

```diff
@@ -1,41 +1,242 @@
 import { gql } from "@apollo/client"
+import { Icon, IconSize } from "@blueprintjs/core"
+import { IconNames } from "@blueprintjs/icons"
 import API from "api"
 import useSearchFilter from "components/advancedSearch/hooks"
 import AdvancedMultiSelect from "components/advancedSelectWidget/AdvancedMultiSelect"
-import { TaskOverlayRow } from "components/advancedSelectWidget/AdvancedSelectOverlayRow"
-import AdvancedSingleSelect from "components/advancedSelectWidget/AdvancedSingleSelect"
+import { AdvancedMultiSelectOverlayTable } from "components/advancedSelectWidget/AdvancedSelectOverlayTable"
 import { getBreadcrumbTrailAsText } from "components/BreadcrumbTrail"
 import TaskTable from "components/TaskTable"
 import { Task } from "models"
 import pluralize from "pluralize"
-import React from "react"
+import React, { useState } from "react"
 import TASKS_ICON from "resources/tasks.png"
 import Settings from "settings"
 
+const taskFields = `
+  ${Task.autocompleteQuery}
+  childrenTasks(query: { pageSize: 1 }) {
+    uuid
+  }
+  parentTask {
+    uuid
+  }
+  descendantTasks {
+    uuid
+  }
+`
+
 const GQL_GET_TASK = gql`
   query ($uuid: String!) {
     task(uuid: $uuid) {
-      uuid
-      shortName
+      ${taskFields}
     }
   }
 `
 
 const GQL_GET_TASKS = gql`
   query ($uuids: [String]) {
     tasks(uuids: $uuids) {
-      uuid
-      shortName
+      ${taskFields}
     }
   }
 `
 
+interface HierarchicalOverlayTableProps {
+  items: any[]
+  selectedItems: any[]
+  handleAddItem?: (...args: unknown[]) => unknown
+  handleRemoveItem?: (...args: unknown[]) => unknown
+}
+
+const HierarchicalOverlayTable = ({
+  items,
+  selectedItems,
+  handleAddItem,
+  handleRemoveItem,
+  ...otherProps
+}: HierarchicalOverlayTableProps) => {
+  const [expandedItems, setExpandedItems] = useState(new Set<string>())
+  const [childrenMap, setChildrenMap] = useState(new Map<string, any[]>())
+
+  const fetchChildren = async task => {
+    const query = gql`
+      query ($query: TaskSearchQueryInput) {
+        taskList(query: $query) {
+          list {
+            uuid
+            shortName
+            longName
+            ascendantTasks {
+              ${taskFields}
+            }
+            childrenTasks {
+              uuid
+            }
+            descendantTasks {
+              uuid
+            }
+            parentTask {
+              childrenTasks {
+                ${taskFields}
+              }
+            }
```

Comment:
This feels odd. You fetch all children of a certain parentTask with the query, then for each child get the parentTask and then all children of this parent. The list you get there will be the same as the result of the outer query, so why request this redundant information?